tiledb/py_arrow_io_impl.h

```diff
@@ -233,6 +233,8 @@ ArrowInfo tiledb_buffer_arrow_fmt(BufferInfo bufferinfo, bool use_list = true) {
     return ArrowInfo("tsu:");
   case TILEDB_DATETIME_NS:
     return ArrowInfo("tsn:");
+  case TILEDB_DATETIME_DAY:
+    return ArrowInfo("tdD");
```

Comment:
The [`tdD` type](https://arrow.apache.org/docs/format/CDataInterface.html#data-type-description-format-strings) is a 32-bit number of days, while `TILEDB_DATETIME_DAY` is 64-bit. You will have to either:

* Narrow the numbers in the data buffer to 32-bit.
* Multiply them by the number of milliseconds in a day, and present them as date64 (`tdm`).

Reply:
I did something similar to the 2nd bullet.